assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -5357,6 +5360,11 @@ public void doPreview_submission(RunData data) {
         state.setAttribute(PREVIEW_SUBMISSION_ASSIGNMENT_REFERENCE, aReference);
         Assignment a = getAssignment(aReference, "doPreview_submission", state);
 
+        String[] groupChoice = params.getStrings("selectedGroups");
+        if (groupChoice != null && groupChoice.length != 0) {
```

Comment:
using isEmpty is more concise `if (groupChoice != null && !groupChoice.isEmpty())`